src/cargo/util/toml/mod.rs

```diff
@@ -2135,7 +2139,16 @@ fn to_dependency_source_id<P: ResolveToPath + Clone>(
             // always end up hashing to the same value no matter where it's
             // built from.
             if manifest_ctx.source_id.is_path() {
-                let path = manifest_ctx.root.join(path);
+                let path = if let Some(base) = orig.base.as_ref() {
+                    lookup_path_base(&base, manifest_ctx.gctx, manifest_ctx.root)
+                        .with_context(|| {
+                            format!("resolving path base for dependency ({name_in_toml})")
+                        })?
+                        .join(path)
+                } else {
+                    // This is a standard path with no prefix.
+                    manifest_ctx.root.join(path)
+                };
```

Comment:
I was hoping this could be taken care of during `normalize_dependencies`, dropping `base`.  I've been moving this code more generally to the model of data transformation happens during `normalize` and `to_real_manifest` is about adapting data types.

We could then do a follow up refactor to unify that with the inheriting dependencies without a base, as the base then is just "workspace"

Reply:
Ok, moved it into `normalize`

Reply:
ok

> We could then do a follow up refactor to unify that with the inheriting dependencies without a base, as the base then is just "workspace"

I've noted this as unresolved issue on #14355